tests/compat.sh

```diff
@@ -278,12 +283,6 @@ filter_ciphersuites()
         # Ciphersuite for GnuTLS
         G_CIPHERS=$( filter "$G_CIPHERS" )
     fi
-
-    # For GnuTLS client -> Mbed TLS server,
-    # we need to force IPv4 by connecting to 127.0.0.1 but then auth fails
-    if is_dtls "$MODE" && [ "X$VERIFY" = "XYES" ]; then
-        G_CIPHERS=""
-    fi
 }
 
```

Comment:
2.28 only wrinkle (because it's about protocol versions before 1.2):

```
            # TLS-RSA-WITH-NULL-SHA256 is a (D)TLS 1.2-only cipher suite,
            # like all SHA256 cipher suites. But Mbed TLS supports it with
            # (D)TLS 1.0 and 1.1 as well. So do ancient versions of GnuTLS,
            # but this was considered a bug which was fixed in GnuTLS 3.4.7.
            # Check the GnuTLS support list to see what the protocol version
            # requirement is for that cipher suite.
            if [ `minor_ver "$MODE"` -ge 3 ] || {
                   [ `minor_ver "$MODE"` -gt 0 ] &&
                   $GNUTLS_CLI --list | grep -q '^TLS_RSA_NULL_SHA256.*0$'
               }
            then
                M_CIPHERS="$M_CIPHERS                           \
                    TLS-RSA-WITH-NULL-SHA256                    \
                    "
                G_CIPHERS="$G_CIPHERS                           \
                    +RSA:+NULL:+SHA256                          \
                    "
            fi
```

When we run outcome analysis, we should treat `TLS-RSA-WITH-NULL-SHA256` as present, since we do expect to run it on the CI. So I think that `compat.sh --list-test-cases` should list it unconditionally (and in fact shouldn't try running `gnutls-cli` at all).

Actually, we should probably do this check differently: always include the cipher suite in the list, but skip it if GnuTLS doesn't support it for that protocol version. That should resolve the `--list-test-cases` output automatically.

Reply:
Tentative fix: https://github.com/gilles-peskine-arm/mbedtls/tree/backport-8586%2Bgnutls_null_sha256-list_always [CI job](https://mbedtls.trustedfirmware.org/job/mbedtls-restricted-release-new/2/)

Reply:
Thanks for noticing, analysing and suggesting a fix! Your fix looks good to me so I'm cherry-picking it.

Reply:
Er, given the logs I think I messed something up, either causing tests not to be executed or spurious test cases to be logged or some naming convention mismatch.

Reply:
Aw. I'm investigating.

Reply:
Ah, no, all of those were already not executed [before](https://mbedtls.trustedfirmware.org/job/mbed-tls-pr-head/job/PR-8956-head/3/execution/node/6696/log/?consoleFull), except for the old-version `+RSA:+NULL:+SHA256`, which are supposed to be executed on the CI so it's good that they're getting reported (and if they're getting executed but not being reported as such, that seems to be part of a larger preexisting issue).